nncf/experimental/common/pruning/operations.py

```diff
@@ -275,6 +284,26 @@ def _get_output_mask(cls, input_masks: List[Optional[PropagationMask]]) -> Optio
                             group.invalidate()
         return output_mask
 
+    @staticmethod
+    def can_propagate_mask(input_masks: List[Optional[PropagationMask]], input_shapes: List[Tuple[int, ...]]) -> bool:
+        """
+        Check that input without a mask does not affect input with a mask.
+
+        :param input_masks: Input masks.
+        :param input_shapes: Input shapes.
+        """
+        none_mask_ind = input_masks.index(None)
+        mask_ind = 0 if none_mask_ind else 1
+        if len(input_shapes[none_mask_ind]) > len(input_shapes[mask_ind]):
```

Comment:
Is it valid case when inputs of element-wise operation have different length of shapes?

Reply:
It's for case when shapes are (1,1,10,10) + (1,10) and mask exists for second input, propagated mask will be incorrect for output berceuse dimension changed